### PR TITLE
Goodwe warning fixes

### PR DESCRIPTION
**Pull request by llagendijk**
Fixed the warnings that you asked me to fix in the Goodwe adapter. Had to break deviceID calculation unfortunately.

### CODE REVIEW

File: hardware/GoodweAPI.cpp [hunk 1]
```diff
@@ -166,11 +166,12 @@ int GoodweAPI::hash(std::string str)
...
 	int i = 0;
-	int c;
+	int c = str[i++];
 	
-	while(c = str[i++])
+	while(c)
 	{
 		hash = ((hash << 5) + hash) + c;
+		c = str[i++];
 	}
```
Comment (gizmocuz) [inline hardware/GoodweAPI.cpp:176]: I still find this a bit strange way todo this
You know the length of the string ( str.size() )
You could use

while (i<str.size())

or even

for (i=0; i<str.size(); i++)

Both look a bit better

File: hardware/GoodweAPI.cpp [hunk 2]
```diff
@@ -442,9 +443,10 @@ void GoodweAPI::ParseDevice(Json::Value device, std::string sStationId, std::str
 	int NodeID = hash(sStationId);
 	int ChildID = hash(sDeviceSerial);
 
-	// reserve childid below 10 for the station
-	if (ChildID < 10)
-		ChildID =+ 10;
+	// reserve childIDs  0 - 10 for the station
+	if ((ChildID >= 0) && (ChildID <= 10)) {
+		ChildID = ChildID + 10;
+	}
```
Comment (gizmocuz) [inline hardware/GoodweAPI.cpp:451]: Could there be negative child id's ? maybe the hash should return an uint32_t ?

File: hardware/GoodweAPI.cpp (hunk 1 above)
Comment (llagendijk) [reply hardware/GoodweAPI.cpp:176]: Good comment. I am still too much C-oriented...
Will include this

File: hardware/GoodweAPI.cpp [hunk 3]
```diff
@@ -168,11 +169,12 @@ int GoodweAPI::hash(std::string str)
 	int i = 0;
```
Comment (gizmocuz) [inline hardware/GoodweAPI.cpp]: could you make it size_t instead of int ;)

Comment (llagendijk) [reply hardware/GoodweAPI.cpp]: I should have known this but missed it :-(

**Comment by gizmocuz** [comment]
Thanks !

**Pull request merged**